### PR TITLE
CPB-15, CPB-43: Create CV model, Update CV model

### PR DESCRIPTION
**Pull request by LV-coding**

### CODE REVIEW

File: backend/apps/cover_letters/models.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from django.utils.translation import gettext_lazy as _
+from apps.cover_letters.storages import OverwriteStorage
+
+
+def content_file_name(instance, filename):
+    file_path = '/'.join(['cv', instance.owner.full_name, filename])
+    return file_path
+
+
+class CoverLetter(models.Model):
```
Comment (pazuzu-ua) [inline backend/apps/cover_letters/models.py]: CV stands for Curriculum Vitae, not a Cover Letter.

File: backend/apps/cover_letters/models.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from apps.candidates.models import Candidate
+from apps.vacancies.models import Vacancy
+from django.utils.translation import gettext_lazy as _
+from apps.cover_letters.storages import OverwriteStorage
+
+
+def content_file_name(instance, filename):
+    file_path = '/'.join(['cv', instance.owner.full_name, filename])
```
Comment (pazuzu-ua) [inline backend/apps/cover_letters/models.py]: "/"             "cv"
Are you using flake8 and blake?

**Review by pazuzu-ua — COMMENTED** [review]
Please, rename the app

**Comment by LV-coding** [comment]
> Please, rename the app

for what? Curriculum Vitae - few people have heard

**Comment by pazuzu-ua** [comment]
> > Please, rename the app
> 
> for what? Curriculum Vitae - few people have heard

First of all, a lot of Europeans know this word. And lot's of educated people throughout US know it. If you don't like it, change to Resume then.
CV stands for Curriculum Vitae (or Resume). Cover Letter has nothing to do with CV. Cover Letter is a Letter which is supplemented to the CV and is required for admission to the university, not only job. You are creating a CV model, not a cover letter model.

File: backend/apps/cover_letters/models.py (hunk 1 above)
Comment (LV-coding) [reply backend/apps/cover_letters/models.py]: fixed

File: backend/apps/resume/admin.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+from django.contrib import admin
+from apps.resume.models import CurriculumVitae
+
+
+@admin.register(CurriculumVitae)
+class CoverLetterAdmin(admin.ModelAdmin):
```
Comment (pazuzu-ua) [inline backend/apps/resume/admin.py]: CurriculumVitaeAdmin

File: backend/apps/resume/apps.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class CoverLettersConfig(AppConfig):
```
Comment (pazuzu-ua) [inline backend/apps/resume/apps.py]: CurriculumVitaeConfig

**Review by pazuzu-ua — APPROVED** [review]
Please, exclude Certificate_CS50.pdf and rename 2 more classes and all is good!

File: backend/apps/resume/models.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+    owner = models.ForeignKey(
+        Candidate,
+        on_delete=models.CASCADE, 
+        related_name="cv_owner",
+        verbose_name=_("CV owner"),
+    )
+    vacancy = models.ManyToManyField(
+        Vacancy,
+        related_name="cv_vacancies",
```
Comment (devoorhees) [inline backend/apps/resume/models.py:44]: related_name="vacancy_cvs"

File: backend/apps/resume/models.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from apps.candidates.models import Candidate
+from apps.vacancies.models import Vacancy
+from django.utils.translation import gettext_lazy as _
+from apps.resume.storages import OverwriteStorage
+
+
+def content_file_name(instance, filename):
+    file_path = "/".join(["cv", instance.owner.full_name, filename])
+    return file_path
```
Comment (devoorhees) [inline backend/apps/resume/models.py]: Rename the function to something like `generate_file_name()` or add a docstring. 
Also you don't need to create a `file_path` variable if you use it only once:
```
return "/".join(["cv", instance.owner.full_name, filename])
```
I guess, there is no need to call `.join()` here, your list is not such large to do this extra joint, you can use f-string instead:
```
return f"cv/{instance.owner.full_name}/{filename}"
```

File: backend/apps/resume/models.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        max_length=2048,
+        blank=True
+    )
+    processed_by_tesseract = models.BooleanField(
+        _("Processed by tesseract"),
+        default=False
+    )
+    created_at = models.DateTimeField(
+        _("Created at"),
+        auto_now=True,
```
Comment (devoorhees) [inline backend/apps/resume/models.py]: auto_now_add=True

File: backend/apps/resume/models.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    )
+    changed_at = models.DateTimeField(
+        _("Last updated"),
+        auto_now_add=True,
```
Comment (devoorhees) [inline backend/apps/resume/models.py]: auto_now=True

File: backend/apps/resume/models.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    class Meta:
+        ordering = ("-id",)
+
+    def __str__(self) -> str:
+        """Return CV owner"""
+        return f"{self.owner} CV"
+
+    def __repr__(self) -> str:
+        """Return debug info for the CV."""
+        return f"<CoverLetter (id={self.id}) - {self.owner}>"
```
Comment (devoorhees) [inline backend/apps/resume/models.py]: It's not a CoverLetter !)

File: backend/apps/resume/models.py [hunk 10]
```diff
@@ -0,0 +1,84 @@
+from django.db import models
+from apps.candidates.models import Candidate
+from apps.vacancies.models import Vacancy
+from django.utils.translation import gettext_lazy as _
+from apps.resume.storages import OverwriteStorage
+
+
+def genarate_file_path(instance, filename):
+    file_path = f"cv/{instance.owner.full_name}/{filename}"
+    return file_path
```
Comment (devoorhees) [inline backend/apps/resume/models.py:10]: No need to create a variable if you use it only once!

**Review by devoorhees — APPROVED** [review]

**Pull request merged**